contributed_definitions/nyaml/NXmanipulator.yaml

```diff
@@ -13,32 +13,124 @@ NXmanipulator(NXobject):
   type(NX_CHAR):
     doc: |
       Type of manipulator, Hexapod, Rod, etc.
-  cryocoolant(NX_BOOLEAN):
+  cryostat(NXactuator):
```

Comment:
I like the idea behind this particular refactoring as it also goes along the line of though deeper base class

Traditionally NIAC could have said: (NXmanipulator) has multiple NXactuators, and what we have here is part of the appdef.

I fully support what you have done is to help appdef writers and appdef users who wish to make slight modifications that they dont have to redefine again all tiny details of a cryostat but can use yours.

Again an argument why there has to be more than black and white either base class or appdef but something mediating in between, I dont buy the argument like "but it worked in the past" yes, maybe because also many of the appdefs from the past are less detailed and narrower in research field covered

Reply:
I mean here it is explicitly stated in the docstring of `NXmanipulator` that it is meant to "describe the use of manipulators in _photoemission_ experiments." So it makes perfect sense to also name the typical sensors and actuators that you would have in a PES experiment.